### PR TITLE
Better platform abstraction for lf_clock_gettime() and _LF_CLOCK

### PR DESCRIPTION
**Pull request by Soroosh129**
This makes _LF_CLOCK, which is platform-dependent, an internal property of the given platform's implementation files.

A consequence of that is a backward-incompatible change to lf_clock_gettime(). Any platform support library in the future would have to change (e.g., the RISCV bare-metal support).

This also makes get_physical_time() more platform independent, mainly by using the new lf_clock_gettime().

### CODE REVIEW

File: org.lflang/src/lib/core/platform/lf_linux_support.c [hunk 1]
```diff
@@ -42,10 +42,10 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 /**
- * Fetch the value of clk_id and store it in tp.
+ * Fetch the value of _LF_CLOCK and store it in tp.
```
Comment (edwardalee) [inline org.lflang/src/lib/core/platform/lf_linux_support.c:45]: What is the return value?

File: org.lflang/src/lib/core/platform/lf_macos_support.c [hunk 2]
```diff
@@ -42,10 +42,10 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 /**
- * Fetch the value of clk_id and store it in tp.
+ * Fetch the value of _LF_CLOCK and store it in tp.
```
Comment (edwardalee) [inline org.lflang/src/lib/core/platform/lf_macos_support.c:45]: Return value?

File: org.lflang/src/lib/core/platform/lf_windows_support.c [hunk 3]
```diff
@@ -154,13 +154,13 @@ int lf_cond_timedwait(_lf_cond_t* cond, _lf_critical_section_t* critical_section
 }
 
 /**
- * Fetch the value of clk_id and store it in tp.
+ * Fetch the value _LF_CLOCK and store it in tp.
```
Comment (edwardalee) [inline org.lflang/src/lib/core/platform/lf_windows_support.c:157]: Return value?

**Review by edwardalee — APPROVED** [review]
Looks good to me! Minor documentation requests...

File: org.lflang/src/lib/core/platform.h [hunk 4]
```diff
@@ -129,9 +129,15 @@ extern int lf_cond_timedwait(lf_cond_t* cond, lf_mutex_t* mutex, long long absol
...
 
 /**
- * Fetch the value of clk_id and store it in tp.
+ * Fetch the value of an internal (and platform-specific) physical clock and 
+ * store it in tp. 
+ * 
+ * Ideally, the underlying platform clock should be monotonic. However, the
+ * core lib tries to enforce monotonicity at higher level APIs (see tag.h). 
+ * Nonetheless, if the underlying clock is not monotonic, it can result in 
+ * identical timestamps being presented to the user programs that use those APIs.
```
[the remark this answers is not in the document]
Comment (Soroosh129) [reply org.lflang/src/lib/core/platform.h:138]: I think you are right. I will fix the description.

**Comment by Soroosh129** [comment]
@efsanesoyer @lsk567 FYI, I have removed timespecs from `platform.h` APIs. Instead, they purely use instant_t. This to some degree reduces the overhead of converting back and forth between `timespec` and `instant_t`, although for UNIX-based systems, a one time translation is needed to use clock APIs. But, for bare-iron, I think no messing around with `timespec` is hopefully needed.

**Comment by lsk567** [comment]
This is very helpful. Hopefully, this can resolve the error regarding `LF_CLOCK` that we are looking at right now. Thanks @Soroosh129 !

**Pull request merged**